### PR TITLE
Resolve 2-way data binding issue in BitCheckBox component when the parameter is not provided (#440)

### LINKED ISSUE #440: Two way data Binding issue in BitCheckBox
consumers can work with value in two ways 
1. consumer can get the value of BitCheckBox every time it gets changed using OnChange and in this scenario we provide the component with full functionality
2. consumer can control the value of BitCheckBox using both IsChecked and IsCheckedChanged and it is necessary to provide them both in order to use the full functionality of the component if the consumer provides us with IsChecked we cannot change the value of the component from inside 

currently, OnChange doesn't work and we have to change our approach to implementing two-way data binding

### PR DESCRIPTION
**Pull request by khashayar-pakkhesal**
this closes #440

### CODE REVIEW

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs [hunk 1]
```diff
@@ -61,6 +61,8 @@ public bool IsIndeterminate
 
         [Parameter] public EventCallback<bool> OnChange { get; set; }
 
+        private bool hasSetIsChecked { get; set; }
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs:64]: It is not a good name for this property, I think it should be named something like "IsCheckedHasValue" or "IsCheckHasBeenSet"

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs (hunk 1 above)
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs:64]: And change this property into a private field

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs [hunk 2]
```diff
@@ -108,5 +110,38 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
 
             await base.OnAfterRenderAsync(firstRender);
         }
+
+        public override Task SetParametersAsync(ParameterView parameters)
+        {
+            foreach (ParameterValue parameter in parameters)
+            {
+                switch (parameter.Name)
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs:118]: The private field holding the state of value of 2-way bound parameter should be reset at the start of this method

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs (hunk 1 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs]: Changed hasSetIsChecked  to isCheckedHasValue

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs (hunk 2 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs:121]: Done

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs (hunk 1 above)
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs]: after some investigation, I think the "IsCheckedHasBeenSet" is a better name. please apply this particular pattern for naming this specific field, which will be applied in the source generator in near future.

Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Checkboxes/BitCheckbox.razor.cs]: Rename hasSetIsChecked to IsCheckedHasBeenSet

**Review by msynk — APPROVED** [review]

**Pull request merged**